cpp/src/KIM_SimulatorModelImplementation.cpp

```diff
@@ -464,7 +464,9 @@ void SimulatorModelImplementation::GetParameterFileDirectoryName(
 #endif
   LOG_DEBUG("Enter  " + callString);
 
-  *directoryName = &parameterFileDirectoryName_;
+  // Convert path to conventional string representation.
+  parameterFileDirectoryNameString_ = parameterFileDirectoryName_.string();
```

Comment:
Other API methods seem do it in a similar fashion, see for example [CollectionsImplementation::GetItemLibraryFileNameByCollectionAndType()](https://github.com/openkim/kim-api/blob/devel/cpp/src/KIM_CollectionsImplementation.cpp#L1107-L1120). These methods also update the value the returned pointer is pointing at only lazily and upon request. 

That being said, I should note that I've mainly added this statement here in order to make really really sure that `parameterFileDirectoryNameString_` and `parameterFileDirectoryName_` will be sync also in the future. Presently, they already are in sync at all times, because `SimulatorModelImplementation::Initialize()` sets both value simultaneously.

Reply:
Actually, there is no need to the string variable now that the `path` object is there.  We can just completely eliminate these `...NameString_` variables and avoid the question.

Reply:
I made some changes to this code now, because I updated the signature of `SharedLibrary::GetParameterFileDirectoryName()` to use `FILESYSTEM::Path`. However, I don't see a way to get rid of the    `...NameString_` variables without breaking the interface of `ModelDriverCreate::GetParameterFileDirectoryName()`, which still requires an internal `std::string` storage.

Reply:
Right!  My fault;  I didn't think carefully enough about this.